src/tiled/flipmapobjects.cpp

```diff
@@ -43,10 +44,37 @@ FlipMapObjects::FlipMapObjects(MapDocument *mapDocument,
 
 void FlipMapObjects::flip()
 {
-    // TODO: Flip them properly as a group
     const auto &objects = mMapObjects;
+
+    //computing objects center
+    QRectF boundaryObjectsRect;
+    for (MapObject *object : objects) {
+        QTransform objectTransform;
+        objectTransform.translate(object->x(), object->y());
+        objectTransform.rotate(object->rotation());
+        objectTransform.translate(-object->x(), -object->y());
+
+        if (!object->cell().isEmpty()) { //computing bound rect for cell
+            QRectF cellRect = QRectF(object->x(),
+                                     object->y(),
+                                     object->width(), -object->height()).normalized();
+            boundaryObjectsRect = boundaryObjectsRect.united(objectTransform.mapRect(cellRect));
+        } else if (!object->polygon().empty()) { //computing bound rect for polygon
+            const QPolygonF &objectPolygon = object->polygon();
+            QTransform polygonToMapTransform;
+            polygonToMapTransform.translate(object->x(),
+                                            object->y());
+            polygonToMapTransform.rotate(object->rotation());
+            boundaryObjectsRect = boundaryObjectsRect.united(polygonToMapTransform.mapRect(QRectF(objectPolygon.boundingRect())));
+        } else { //computing bound rect for other
+            boundaryObjectsRect = boundaryObjectsRect.united(objectTransform.mapRect(object->bounds()));
+        }
+    }
+    QPointF objectsCenter = boundaryObjectsRect.center();
```

Comment:
Since the list of objects is not going to change after the `FlipMapObjects` instance is created, please move the calculation of the center to the constructor.